osconfig_tests/test_suites/package_management/package_management_test_data.go

```diff
@@ -58,39 +58,41 @@ func addCreateOsConfigTest(pkgTestSetup []*packageManagementTestSetup, testProje
+			for _, image := range debianImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, osconfigserver.BuildAptPackageConfig(pkgs, nil, nil), nil, nil, nil, nil)
+				setup := NewPackageManagementTestSetup(image, fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), testName, "", oc, nil, nil, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
+			}
 		case "centos":
-			image = centosImage
-			pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
-			oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
+			for _, image := range centosImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
+				setup := NewPackageManagementTestSetup(image, fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), testName, "", oc, nil, nil, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
+			}
 		case "rhel":
-			image = rhelImage
-			pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
-			oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
+			for _, image := range rhelImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
+				setup := NewPackageManagementTestSetup(image, fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), testName, "", oc, nil, nil, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
+			}
 		case "windows":
-			image = windowsImage
-			pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
-			oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, nil, osconfigserver.BuildGooPackageConfig(pkgs, nil, nil), nil, nil)
+			for _, image := range windowsImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, nil, osconfigserver.BuildGooPackageConfig(pkgs, nil, nil), nil, nil)
+				setup := NewPackageManagementTestSetup(image, fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), testName, "", oc, nil, nil, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
```

Comment:
it looks like the `setup :=` and `pkgTestSetup :=` lines are duplicated for each case. Can they be moved after the case statements? Alternatively, you could factor our these cases into separate functions if you think that would be cleaner.

Reply:
moving it to separate function since moving it out of switch would require to create a new slice for each platform and in the end append each slice to the common pkgtestSetup slice(too much work).

Refactored it out to a separate function.